handlers/main-macos.yml

```diff
@@ -8,20 +8,23 @@
 # to finish and if it's in progress for a longer time, bootstrap fails. We use the old
 # unload/load combo because they actually wait.
 - name: Unload datadog-agent service
-  command: "launchctl unload -wF {{ datadog_macos_system_plist_file_path }}"
-  become: yes
-  check_mode: no
+  command: launchctl unload -wF {{ datadog_macos_system_plist_file_path }}
+  become: true
+  check_mode: false
   when: datadog_enabled and not ansible_check_mode and ansible_facts.os_family == "Darwin"
+  changed_when: true
```

Comment:
Actually, for tasks where Ansible cannot tell by itself if something changed (basically command and shell tasks), ansible-lint force us to add the changed_when directive, to indicate a condition when the task must be considered as changed. Here I think that each time the `when` condition is met, the task will change something (unload the agent). From what I understand of this task it should either be skipped when the `when` condition is not met or reported as `changed`

Reply:
Right, it's true that the handlers should always have `changed_when: true`, because they actually will restart the Agent. We should just make sure we're triggering (or not triggering) the handlers in the correct situations.